backend/matching-service/src/services/rabbitmq.service.ts

```diff
@@ -178,6 +182,7 @@ class RabbitMQConnection {
                     logger.info(`[Waiting-Queue] Match found: ${directMatch.content.toString()}`)
                     this.channel.ack(directMatch)
                     await this.removeIfEmptyQueue(destinationQueue)
+                    matchedUser = directMatch
```

Comment:
I think its safer to assign matched user before ack-ing the message in the queue, to prevent issues (although unlikely) such as failing to assign match to the variable matchedUser. Same for the rest below.

Reply:
If the message fails to acknowledge, the catch block will intercept the error, and thus avoid the update of data into DB, since the update is triggered as the last step within the callback.

Reply:
sure, i've reordered the code 😆